src/ui/windows/TogglDesktop/TogglDesktop.Tests/TimelineConvertTimeEntriesToBlocksTests.cs

```diff
@@ -0,0 +1,170 @@
+        }
+
+        //Time entries scheme:
+        //  |
+        //  ||
+        //   |
+        //  ||
+        //  |
+        private static object[] TestCase1()
+        {
+            var timeEntryList = new List<Toggl.TogglTimeEntryView>()
+            {
+                new Toggl.TogglTimeEntryView()
+                    {GUID = "1", Started = _selectedDateUnix + 0, Ended = _selectedDateUnix + 100},
+                new Toggl.TogglTimeEntryView()
+                    {GUID = "2", Started = _selectedDateUnix + 50, Ended = _selectedDateUnix + 200},
+                new Toggl.TogglTimeEntryView()
+                    {GUID = "3", Started = _selectedDateUnix + 150, Ended = _selectedDateUnix + 200}
+            };
+            var scaleMode = 0;
+            return new object[]
+            {
+                timeEntryList,
+                null,
+                scaleMode,
+                _selectedDate,
+                new List<(string Guid, double VerticalOffset, double Height, double HorizontalOffset, bool IsOverlapping)>()
+                {
+                    (timeEntryList[0].GUID, TimeToHeight(timeEntryList[0].Started - _selectedDateUnix, scaleMode),
+                        TimeToHeight(timeEntryList[0].Ended - timeEntryList[0].Started, scaleMode), 0, true),
+                    (timeEntryList[1].GUID, TimeToHeight(timeEntryList[1].Started - _selectedDateUnix, scaleMode),
+                        TimeToHeight(timeEntryList[1].Ended - timeEntryList[1].Started, scaleMode), 0 + _timeEntryOffset, true),
+                    (timeEntryList[2].GUID, TimeToHeight(timeEntryList[2].Started - _selectedDateUnix, scaleMode),
+                        TimeToHeight(timeEntryList[2].Ended - timeEntryList[2].Started, scaleMode), 0, true)
+                }
+            };
+        }
+
+        //Time entries scheme:
+        //  |
+        //  ||
+        //  |||
+        //    |
+        private static object[] TestCase2()
+        {
+            var timeEntryList = new List<Toggl.TogglTimeEntryView>()
+            {
+                new Toggl.TogglTimeEntryView()
+                    {GUID = "1", Started = _selectedDateUnix + 0, Ended = _selectedDateUnix + 150},
+                new Toggl.TogglTimeEntryView()
+                    {GUID = "2", Started = _selectedDateUnix + 50, Ended = _selectedDateUnix + 150},
+                new Toggl.TogglTimeEntryView()
+                    {GUID = "3", Started = _selectedDateUnix + 100, Ended = _selectedDateUnix + 200}
+            };
+            var scaleMode = 1;
+            return new object[]
+            {
+                timeEntryList,
+                null,
+                scaleMode,
+                _selectedDate,
+                new List<(string Guid, double VerticalOffset, double Height, double HorizontalOffset, bool IsOverlapping)>()
+                {
+                    (timeEntryList[0].GUID, TimeToHeight(timeEntryList[0].Started - _selectedDateUnix, scaleMode),

```

Comment:
This part of the tests is quite hard to read.
My suggestion to improve the readability:
1. Extract constants, like T1 = 0, T2 = 50, T3 = 100, T4 = 150, T5 = 200.
2. Use theses constants in your expected blocks list, e.g. `TimeToHeight(T1, scaleMode)` is easier to read than `TimeToHeight(timeEntryList[0].Started - _selectedDateUnix, scaleMode)`.